src/System.IO.Compression/src/System/IO/Compression/Crc32Helper.cs

```diff
@@ -449,16 +453,46 @@ internal static class Crc32Helper
             0x6EAB0882u, 0xA201081Cu, 0xA8C40105u, 0x646E019Bu, 0xEAE10678u,
             0x264B06E6u
         };
-
+        private static WorkerType GetCrcType()
+        {
+            try
+            {
+                // Make a P/Invoke into zlib crc32 to ensure we're able to find and use it.
+                // If we are, then use zlib.
+                byte[] buffer = new byte[1];
+                Interop.zlib.crc32(0,buffer,0,0);
```

Comment:
Since we have an overload that takes a ```byte*```, could we just pass null?  I think crc32 supports that.  Could you just pass an address of a single byte on the stack rather than allocating a new array?